doc/docs/master/how-tos/developer-workflow/index.md

```diff
@@ -0,0 +1,32 @@
+# Developer Workflow
+
+In general, the developer workflow for Pachyderm involves adding 
+data to a versioned data repository and adding a pipeline that 
+reads from that data repository to execute your code against that data. 
+Both the data and pipeline can be iterated on independenly with Pachyderm
+handling the code execution according to the pipeline specfication.
+The workflow steps are shown below. 
```

Comment:
This might be slightly pedantic, but the above paragraph implies to me that most use cases in Pachyderm are adequately covered by using a single repo + a single pipeline, which isn't true